src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -798,14 +904,212 @@ private void PlayWithCurrentSettingPressed()
+            MicrobeSpeciesOrganelleStatistics;
+
+        /// <summary>
+        ///   The current generation auto-evo has evolved
+        /// </summary>
+        public int CurrentGeneration;
+
+        /// <summary>
+        ///   Total used auto-evo time
+        /// </summary>
+        public TimeSpan TotalTimeUsed;
+
+        public AutoEvoExploringToolWorld(IAutoEvoConfiguration configuration)
+        {
+            AutoEvoConfiguration = configuration.Clone();
+            GameProperties = GameProperties.StartNewMicrobeGame(new WorldGenerationSettings
+                { AutoEvoConfiguration = AutoEvoConfiguration });
+            SpeciesHistoryList = new List<Dictionary<uint, Species>>();
+            PatchHistoryList = new List<Dictionary<int, PatchSnapshot>>();
+            RunResultsList = new List<LocalizedStringBuilder>();
+            CurrentGeneration = 0;
+            TotalTimeUsed = TimeSpan.Zero;
+
+            RunResultsList.Add(new LocalizedStringBuilder());
+            SpeciesHistoryList.Add(new Dictionary<uint, Species>
+            {
+                {
+                    GameProperties.GameWorld.PlayerSpecies.ID,
+                    (Species)GameProperties.GameWorld.PlayerSpecies.Clone()
+                },
+            });
+
+            PatchHistoryList.Add(GameProperties.GameWorld.Map.Patches.ToDictionary(pair => pair.Key,
+                pair => (PatchSnapshot)pair.Value.CurrentSnapshot.Clone()));
+
+            PatchesCount = GameProperties.GameWorld.Map.Patches.Count;
+
+            MicrobeSpeciesOrganelleStatistics =
+                new Dictionary<OrganelleDefinition, (double Percentage, double Average)>();
+
+            foreach (var organelle in SimulationParameters.Instance.GetAllOrganelles())
+            {
+                MicrobeSpeciesOrganelleStatistics.Add(organelle, (0, 0));
+            }
+
+            UpdateWorldStatistics();
+        }
+
+        public int TotalSpeciesCount { get; private set; }
+
+        public int CurrentSpeciesCount { get; private set; }
+
+        public int PatchesCount { get; }
+
+        public double PatchSpeciesCountAverage { get; private set; }
+
+        public double PatchSpeciesCountStandardDeviation { get; private set; }
+
+        public long TotalPopulation { get; private set; }
+
+        public double MicrobeSpeciesAverageHexSize { get; private set; }
+
+        public void UpdateWorldStatistics()
+        {
+            TotalSpeciesCount = (int)SpeciesHistoryList.Last().Values.Max(s => s.ID);
+            CurrentSpeciesCount = SpeciesHistoryList.Last().Values.Count;
+            PatchSpeciesCountAverage = GameProperties.GameWorld.Map.Patches.Values.Average(p => p.SpeciesInPatch.Count);
+            PatchSpeciesCountStandardDeviation = Math.Sqrt(GameProperties.GameWorld.Map.Patches.Values
+                .Select(p => p.SpeciesInPatch.Count)
+                .Average(c => (c - PatchSpeciesCountAverage) * (c - PatchSpeciesCountAverage)));
```

Comment:
I see this standard deviation calculation quite a bit. I think it might be a bit nicer if MathUtils was updated to have a helper method for calculating the standard deviation and average.

Reply:
I actually tried to, but I failed because int, uint, float and double can be calculated.

Reply:
I think the method would be just about 5-ish lines? So copy-pasting that 3 times is not too bad. I've had to use the same approach for other math helpers as C# sadly doesn't have a generic numeric support.